l3build-aux.lua

```diff
@@ -68,11 +68,11 @@ end
 -- Do some subtarget for all modules in a bundle
 function call(dirs, target, opts)
   -- Turn the option table into a string
-  local opts = opts or options
+  local opts = opts or Opts
```

Comment:
This breaks compatibility with lots of existing code (including latex2e and latex3) and the use of uppercase variables violates l3build's coding conventions. Also why?

Reply:
No need to break any `build.lua`. The global `options` seen in `build.lua` and the global `options` used by l3build need not be the same object, they just have to share some contents at some time. The question is: can `build.lua` change the `options` safely? 

Uppercasing the first letter is a convention used by linters. This is how we can make the difference between local and global variables. 

Of course this does not apply to `build.lua` which is completely independent.

Is there any other place where things could break ?

Reply:
They don't have to be the same object, but separating the only to make them basically equivalent again seems kind of pointless.

Also yes, they can be changed safely (Exceptions: Changing `target` to `help` or `version` wouldn't work, but that would be kind of pointless anyway since build.lua can just call `help` or `version`) (Also yes, there are build.lua files using this feature.)

I'm a bit confused why a Lua linter would default to such settings since I'm not aware of any Lua libraries actually using that convention, but IMO a linter should be configured to be compatible with the project specific coding conventions and not the reverse, especially when the coding conventions affect documented external interfaces.

In which way is `build.lua` "completely independent"? It does share a bunch of global state with the rest of l3build, it is allowed to overwrite a lot of internal functions and is loaded without any isolation. That might not be how you would design such a system (it's certainly not how I would design such a system if I would start from scratch) but for `l3build` and any reasonably compatible system I would describe `build.lua` more as "closely coupled".

Reply:
> In which way is `build.lua` "completely independent"? It does share a bunch of global state with the rest of l3build, it is allowed to overwrite a lot of internal functions and is loaded without any isolation. That might not be how you would design such a system (it's certainly not how I would design such a system if I would start from scratch) but for `l3build` and any reasonably compatible system I would describe `build.lua` more as "closely coupled".

We started with quite a simple set up for just `latex3`, but as I've said expecting `texlua build.lua` as the lead-off. Supporting the LaTeX kernel requires quite a lot of customisation, and I didn't necessarily add it in the most ideal way initially. Some of that has been tided up (e.g. the format-building code hooks), but there are doubtless more places we could adjust.

One might argue that we could drop quite a lot of stuff: as a build system, stability isn't as important as it is for general code.